### PR TITLE
implemented VR532

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
This PR will add the VR532 input validation rule and add it to the mandatory input tests

## References

* #193

### CODE REVIEW

**Review by bemwen — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs [hunk 1]
```diff
@@ -26,30 +26,20 @@ public static void ThrowExceptionIfRequiredPropertyIsNull(ChargeCommand chargeCo
             if (chargeCommand == null) throw new ArgumentNullException(nameof(chargeCommand));
             if (string.IsNullOrWhiteSpace(chargeCommand.CorrelationId)) throw new ArgumentException(chargeCommand.CorrelationId);
 
-            CheckDocument(chargeCommand.Document);
             CheckChargeOperation(chargeCommand.ChargeOperation);
+            CheckDocument(chargeCommand.Document);
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs:30]: Just curious: Why did you change the order? I would think that the document naturally comes before the operation.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeOwnerIsRequiredValidationRule.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public class ChargeOwnerIsRequiredValidationRule : IValidationRule
+    {
+        private readonly ChargeCommand _chargeCommand;
+
+        public ChargeOwnerIsRequiredValidationRule(ChargeCommand chargeCommand)
+        {
+            _chargeCommand = chargeCommand;
+        }
+
+        public bool IsValid => !string.IsNullOrWhiteSpace(_chargeCommand.ChargeOperation.ChargeOwner);
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeOwnerIsRequiredValidationRule.cs:28]: I guess that the `OrWhiteSpace` part is correct. It would be nice with a unit test for that scenario.

**Review by HenrikSommer — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs [hunk 3]
```diff
@@ -26,30 +26,20 @@ public static void ThrowExceptionIfRequiredPropertyIsNull(ChargeCommand chargeCo
...
             if (chargeOperation == null) throw new ArgumentNullException(nameof(chargeOperation));
-            if (string.IsNullOrWhiteSpace(chargeOperation.ChargeOwner)) throw new ArgumentException(chargeOperation.ChargeOwner);
             if (string.IsNullOrWhiteSpace(chargeOperation.ChargeName)) throw new ArgumentException(chargeOperation.ChargeName);
             if (string.IsNullOrWhiteSpace(chargeOperation.ChargeDescription)) throw new ArgumentException(chargeOperation.ChargeDescription);
         }
 
         private static void CheckDocument(Document document)
         {
             if (document == null) throw new ArgumentNullException(nameof(document));
-            if (string.IsNullOrWhiteSpace(document.Id)) throw new ArgumentException(document.Id);
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs:44]: I don't think that you've addressed the document.Id check that you're removing here.

**Review by BjarkeMeier — APPROVED** [review]
Good job. Left a few comments.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs (hunk 1 above)
Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs]: It was a mistake, i reinserted the line myself and choose the wrong location :)

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeOwnerIsRequiredValidationRule.cs (hunk 2 above)
Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeOwnerIsRequiredValidationRule.cs:28]: image

I would think I have covered it in this test, could you help me find what I am missing?

Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeOwnerIsRequiredValidationRule.cs:28]: image

I would think all cases were covered, which one's are missing?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs (hunk 3 above)
Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/ChargeCommandNullChecker.cs]: You are right. I have reinserted it. Funny that I cant find a input validation rule that would ensure the document has a Id, I'm gonna ask @prtandrup about it.
